giskard/client/giskard_client.py

```diff
@@ -180,7 +184,40 @@ def get_project(self, project_key: str) -> Project:
         response = self._session.get("project", params={"key": project_key}).json()
         return Project(self._session, response["key"], response["id"])
 
-    def create_project(self, project_key: str, name: str, kernel_name: str, description: str = None) -> Project:
+    def initialize_kernel(self):
+        python_version = f"{sys.version_info[0]}.{sys.version_info[1]}"
+        base_kernel_name = f"{gethostname()}-{python_version}"
+        kernel_name = f"{base_kernel_name}-{uuid.uuid4()}"
+
+        kernels = self._session.get("kernels").json()
+        frozen_dependencies = [f"{dist.name}=={dist.version}" for dist in importlib_metadata.distributions()]
+
+        matching_kernels = [
+            kernel
+            for kernel in kernels
+            if kernel["name"].startswith(base_kernel_name)
+            and kernel["pythonVersion"] == python_version
+            and set(frozen_dependencies).issubset(set(kernel["frozenDependencies"].split("\n")))
+        ]
+
+        if len(matching_kernels) > 0:
+            return matching_kernels[0]["name"]
+
+        self._session.post(
```

Comment:
Should we check the result here? Cuz for "PROCESS" kernel, backend will create an env as well. This can fail sometimes

Reply:
I think this object is doing it automatically, contrary to request ?

Reply:
Maybe we should change backend, to make start install deps and freeze if needed ?
We could make it so it save in DB the kernel, and then immediatly launch an async task in job manager to do install + freeze (+start ?) if needed ?

Reply:
(unrelated to this PR)

Reply:
> I think this object is doing it automatically, contrary to request ?

Yes, you are right. So it will raise the exception in case of failed creation, right?